packages/features/ee/billing/service/TeamSubscriptionEventHandler.ts

```diff
@@ -0,0 +1,66 @@
+import logger from "@calcom/lib/logger";
+import { TeamRepository } from "@calcom/lib/server/repository/team";
+
+import type { IBillingRepository, IBillingRepositoryUpdateArgs } from "../repository/IBillingRepository";
+import { IBillingRepositoryCreateArgs, Plan } from "../repository/IBillingRepository";
+import { BillingRecord } from "../repository/IBillingRepository";
+import { InternalTeamBilling } from "../teams/internal-team-billing";
+
+type TSubscriptionUpdate = Omit<IBillingRepositoryUpdateArgs, "id">;
+
+export class TeamSubscriptionEventHandler {
+  constructor(
+    private readonly billingRepository: IBillingRepository,
+    private readonly teamRepository: TeamRepository
+  ) {
+    this.billingRepository = billingRepository;
+    this.teamRepository = teamRepository;
+  }
+
+  async handleUpdate(subscription: Omit<IBillingRepositoryCreateArgs, "teamId" | "planName">) {
+    const log = logger.getSubLogger({ prefix: ["TeamSubscriptionEventHandler.handleUpdate"] });
+    const { subscriptionId } = subscription;
+    // First see if the subscription is already in the billing table
+    const teamSubscriptionInDb = await this.billingRepository.getBySubscriptionId(subscriptionId);
+
+    // If the subscription doesn't exist in the DB, migrate it
+    if (!teamSubscriptionInDb) {
```

Comment:
If the subscription is not in the DB then take this opportunity to migrate the subscription to the billing tables.